### ISSUE TITLE
"Unexpected colon" warning in constructor of class nested inside class template

### ISSUE THREAD

**Issue by troiganto** [uncrustify/uncrustify#2302]
Affected version: Uncrustify 0.69.

Input:
```cpp
template<class T>
class Foo<T>::Baz {
  Baz() noexcept
    : i(0) // does not get aligned!
  {}
};
```
Expected utput:
```cpp
template<class T>
class Foo<T>::Baz {
Baz() noexcept
	: i(0) // does not get aligned!
{
}
};
```
Actual output:
```cpp
template<class T>
class Foo<T>::Baz {
Baz() noexcept
: i(0)     // does not get aligned!
{
}
};
```

Uncrustify also outputs the following warning:
```
c.cpp:4 unexpected colon in col 5 n-parent=NONE c-parent=NONE l=1 bl=1
```
This only happens in conjunction with `noexcept`. Without it, the colon gets parsed correctly as introducing a member initializer list.

[Debug file](https://github.com/uncrustify/uncrustify/files/3217703/uncrustify.debug.txt)

As far as I can tell, the uncrustify chokes on the ~~`Foo<T>::Bar`~~ `Foo<T>::Baz` portion and does not parse `Bar` as a class name. This only happens if the part before it is a template.

In any case, thank you for your time and this amazing tool!

EDIT: s/Bar/Baz/

**Comment by guy-maurel** [comment]
should it not be:
```.cpp
template<class T>
class Foo<T>::Baz {
  Baz(int i) noexcept
    : i(0) // does not get aligned!
  {}
};
```
which Bar?
Remember, uncrustify is not a compiler

**Comment by troiganto** [comment]
Hi @guy-maurel, thank you for your quick response. `Foo<T>::Bar` was a typo on my part. I just double-checked my local code and it does use `Baz` consistently.

The example that you give exhibits the same problem as the original; in particular, it still produces the warning "unexpected colon". Actually adding a class `Foo<T>` and a class member `Baz::i` to make this example "complete" doesn't solve the issue either. FWIW, my original code was a bit more complicated. This here is simply a minimal example that still exhibits the problem.

**Comment by guy-maurel** [comment]
PR #2304 is a bugfix proposal.
Please test it and report.

**Comment by troiganto** [comment]
Hi @guy-maurel, thanks for the speedy fix! It does solve my specific issue, but I think there still are some issues when using non-default settings.

Custom uncrustify settings:
```
nl_class_colon  = force
nl_constr_colon = remove
```
Example code:
```cpp
template<class T>
class Foo<T>::Baz {
  Baz() noexcept : i(0) // gets split, but shouldn't
  {}
};
```
Expected output:
```cpp
template<class T>
class Foo<T>::Baz {
Baz() noexcept : i(0)   // gets split, but shouldn't
{
}
};
```
Actual output:
```cpp
template<class T>
class Foo<T>::Baz {
Baz() noexcept :
	i(0)            // gets split, but shouldn't
{
}
};
```

[Debug file](https://github.com/uncrustify/uncrustify/files/3231330/debug-output.txt)

Please tell me if I should open another issue for this or do any more testing.

**Comment by guy-maurel** [comment]
Yes. Open a new issue as you find one.

**Comment by troiganto** [comment]
Done! Feel free to close this issue once the PR is merged. And again, thank you for your quick help! :slightly_smiling_face:
